### PR TITLE
Web: use raw data in `DeviceEvent::MouseMotion` when using `CursorGrabMode::Confined`

### LINKED ISSUE #3547: Web: use raw mouse input when locking pointer
Currently when using `Window::set_cursor_grab(CursorGrabMode::Locked)` on Web mouse movement will contain mouse acceleration. AFAIU other backends send raw mouse events with mouse acceleration disabled, so Web should follow suit.

This would use [`Element.requestPointerLock` with the `unadjustedMovement` options](https://developer.mozilla.org/en-US/docs/Web/API/Element/requestPointerLock#unadjustedmovement), which is notable only supported by Chrome and even there not on Linux.

### PR DESCRIPTION
**Pull request by daxpedda**
This uses [`Element.requestPointerLock` with the `unadjustedMovement` options](https://developer.mozilla.org/en-US/docs/Web/API/Element/requestPointerLock#unadjustedmovement), which is notable only supported by Chrome and even there not on Linux.

I also updated the documentation to properly reflect the current state.

Needs entry in #2875.
Fixes #3547.

### CODE REVIEW

File: src/changelog/unreleased.md [hunk 1]
```diff
@@ -44,6 +44,9 @@ changelog entry.
 
 - Add `ActiveEventLoop::create_proxy()`.
 - On Web, implement `Error` for `platform::web::CustomCursorError`.
+- On Web, add `ActiveEventLoopExtWeb::is_cursor_lock_raw()` to determine if
+  `DeviceEvent::MouseMotion` is returning raw data, not OS accelerated, when using
+  `CursorGrabMode::Confined`.
```
Comment (kchibisov) [inline src/changelog/unreleased.md:49]: it's `Locked`.

File: src/changelog/unreleased.md [hunk 2]
```diff
@@ -69,6 +72,8 @@ changelog entry.
 - Change signature of `EventLoop::run_app`, `EventLoopExtPumpEvents::pump_app_events` and
   `EventLoopExtRunOnDemand::run_app_on_demand` to accept a `impl ApplicationHandler` directly,
   instead of requiring a `&mut` reference to it.
+- On Web, `CursorGrabMode::Confined` now lets `DeviceEvent::MouseMotion` return raw data, not OS
```
Comment (kchibisov) [inline src/changelog/unreleased.md:75]: locked.

File: src/event.rs [hunk 3]
```diff
@@ -467,6 +467,21 @@ pub enum DeviceEvent {
     ///
     /// This represents raw, unfiltered physical motion. Not to be confused with
     /// [`WindowEvent::CursorMoved`].
+    ///
+    /// ## Platform-specific
+    ///
+    /// **Web:** Always returns OS accelerated data unless [`CursorGrabMode::Confined`] is used depending on browser support, see
```
Comment (kchibisov) [inline src/event.rs:473]: locked, and delta is unaccelerated.

File: src/window.rs [hunk 4]
```diff
@@ -1478,6 +1478,23 @@ impl Window {
     ///     .unwrap();
     /// # }
     /// ```
+    ///
+    /// ## Platform-specific
+    ///
+    /// **Web:** Always returns an [`ExternalError::NotSupported`] with
+    /// [`CursorGrabMode::Confined`]. [`DeviceEvent::MouseMotion`] might return OS accelerated data
```
Comment (kchibisov) [inline src/window.rs:1485]: locked.

**Review by kchibisov — CHANGES_REQUESTED** [review]
I'd assume that you mean `Locked` pointer, which is the one that stays in place.

Also, could you change the cursor position when the cursor is locked, since it's other API that usually works with locked cursor?

And I'd assume there's no way to confine cursor at all on web?

**Comment by daxpedda** [comment]
> Also, could you change the cursor position when the cursor is locked, since it's other API that usually works with locked cursor?

You can't change the cursor position on Web. What other API?
I'm not sure what you mean with that question.

> And I'd assume there's no way to confine cursor at all on web?

Not AFAIK.

**Comment by kchibisov** [comment]
>You can't change the cursor position on Web. What other API?
I'm not sure what you mean with that question.

Locked cursor is usually works in pair with `set_cursor_position` API, since otherwise it's sort of useless, unless you completely hide the cursor and draw everything yourself, which could also be an option.

On Wayland, for example, where manipulating with cursor is also prohibited, the _locked pointer_ has its own method to set its position, but it only works when you have the cursor lock present.

It's not really mandatory, it's just changes how you write things basically.

File: src/event.rs (hunk 3 above)
Comment (daxpedda) [reply src/event.rs]: It says "accelerated data **unless**".
I removed the negation which makes it easier to understand though.

Thanks!

File: src/window.rs (hunk 4 above)
Comment (daxpedda) [reply src/window.rs:1485]: This one is actually correct!

**Comment by daxpedda** [comment]
> Locked cursor is usually works in pair with `set_cursor_position` API, since otherwise it's sort of useless, unless you completely hide the cursor and draw everything yourself, which could also be an option.

Ah I see!
Yes, on Web the cursor is hidden, so its not a problem.
The position of the cursor remains the same as the position it was when it was locked.

Is the cursor not hidden in other backends when locked?
If not I should document this for Web.

**Comment by kchibisov** [comment]
>Is the cursor not hidden in other backends when locked?
If not I should document this for Web.

it's generally not hidden and users tend to hide it themselves.

**Comment by daxpedda** [comment]
> > Is the cursor not hidden in other backends when locked?
> > If not I should document this for Web.
> 
> it's generally not hidden and users tend to hide it themselves.

Ah, `CursorGrabMode::Locked` already says:
> There's no guarantee that the cursor will be hidden. You should hide it by yourself if you want to do so.

I removed some duplicate documentation now that I see `CursorGrabMode` includes it already.

**Review by kchibisov — APPROVED** [review]

**Pull request merged**